### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by bwohlberg**
Various changes, primarily related to documentation, including discussion of supported Python version.

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -26,11 +26,11 @@ Installing a Development Version
 
       git clone --recurse-submodules git@github.com:<username>/scico.git
 
-4. Create a conda environment using Python 3.8 or later, e.g.:
+4. Create a conda environment using Python 3.10 or later, e.g.:
 
    ::
 
-      conda create -n scico python=3.9
+      conda create -n scico python=3.11
```
Comment (Michael-T-McCann) [inline docs/source/contributing.rst:33]: I suggest `conda create -n temp "python>=3.10,<3.13"` as a more future-proof option.

File: docs/source/install.rst [hunk 2]
```diff
@@ -5,20 +5,26 @@ Installing SCICO
 
 SCICO requires Python version 3.8 or later. (Version 3.10 is
 recommended as it is the version under which SCICO has been most
-thoroughly tested.) It is supported on both Linux and MacOS, but is
-not currently supported on Windows due to the limited support for
-``jaxlib`` on Windows. However, Windows users can use SCICO via the
-`Windows Subsystem for Linux
+thoroughly tested, and is the minimum supported Python version for
+the most recent versions of JAX.) It is supported on both Linux and
```
Comment (Michael-T-McCann) [inline docs/source/install.rst:9]: "It is supported..." -> "SCICO is supported"

File: misc/conda/README.rst [hunk 3]
```diff
@@ -19,11 +19,11 @@ To install miniconda in ``/opt/conda`` do
    ./install_conda.sh -y /opt/conda
 
 
-To create a conda environment called ``py39`` with default Python version (3.9) and without GPU support
+To create a conda environment called ``py310`` with default Python version (3.10) and without GPU support
 
 ::
 
-   ./make_conda_env.sh -y -e py39
+   ./make_conda_env.sh -y -e py310
```
Comment (Michael-T-McCann) [inline misc/conda/README.rst:26]: Maybe this env should be called `scico` to make it more consistent with `docs/source/contributing.rst`

File: docs/source/contributing.rst (hunk 1 above)
Comment (Michael-T-McCann) [reply docs/source/contributing.rst:33]: Or maybe `conda create -n scico python=3.10` if that really is the version that is most thoroughly tested.

File: scico/_version.py [hunk 4]
```diff
@@ -43,14 +41,7 @@ def variable_assign_value(path: str, var: str) -> Any:
         try:
             # See http://stackoverflow.com/questions/2058802
```
Comment (Michael-T-McCann) [inline scico/_version.py:42]: Could link directly to the answer used, which I think is https://stackoverflow.com/a/30471662

**Review by Michael-T-McCann — APPROVED** [review]
Added optional suggestions.

File: docs/source/install.rst (hunk 2 above)
Comment (bwohlberg) [reply docs/source/install.rst:9]: Done.

File: docs/source/contributing.rst (hunk 1 above)
Comment (bwohlberg) [reply docs/source/contributing.rst:33]: Switched to 3.12, which is the current stable version, and also the version tested in scico CI.

File: misc/conda/README.rst (hunk 3 above)
Comment (bwohlberg) [reply misc/conda/README.rst:26]: Done.

**Pull request merged**